supervisor.go

```diff
@@ -0,0 +1,193 @@
+	supervisorHistoryAllPathPrefix   = "druid/indexer/v1/supervisor/history"
+	supervisorHistoryPathPrefix      = "druid/indexer/v1/supervisor/:supervisorId/history"
+	supervisorSuspendPathPrefix      = "druid/indexer/v1/supervisor/:supervisorId/suspend"
+	supervisorSuspendAllPathPrefix   = "druid/indexer/v1/supervisor/suspendAll"
+	supervisorResumePathPrefix       = "druid/indexer/v1/supervisor/:supervisorId/resume"
+	supervisorResumeAllPathPrefix    = "druid/indexer/v1/supervisor/resumeAll"
+	supervisorResetPathPrefix        = "druid/indexer/v1/supervisor/:supervisorId/reset"
+	supervisorTerminatePathPrefix    = "druid/indexer/v1/supervisor/:supervisorId/terminate"
+	supervisorTerminateAllPathPrefix = "druid/indexer/v1/supervisor/terminateAll"
+	supervisorShutdownPathPrefix     = "druid/indexer/v1/supervisor/:supervisorId/shutdown"
+)
+
+// SupervisorService is a service that submits ingestion tasks to druid supervisor API.
+type SupervisorService struct {
+	client *Client
+}
+
+// CreateSupervisorResponse is a response object of Druid SupervisorService's SubmitTask method
+type CreateSupervisorResponse struct {
+	SupervisorId string `json:"id"`
+}
+
+// TerminateSupervisorResponse is a response object of Druid SupervisorService's Terminate method
+type TerminateSupervisorResponse struct {
+	SupervisorId string `json:"id"`
+}
+
+type GetStatusResponsePayload struct {
+	Datasource      string `json:"dataSource"`
+	Stream          string `json:"stream"`
+	State           string `json:"state"`
+	Partitions      int    `json:"partitions"`
+	Replicas        int    `json:"replicas"`
+	DurationSeconds int    `json:"durationSeconds"`
+}
+
+// GetStatusResponse is a response object of Druid SupervisorService's GetStatus method
+type GetStatusResponse struct {
+	SupervisorId   string                   `json:"id"`
+	GenerationTime time.Time                `json:"generationTime"`
+	Payload        GetStatusResponsePayload `json:"payload"`
+}
+
+// CreateOrUpdate submits an ingestion specification to druid Supervisor API with a pre-configured druid client
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#create-or-update-a-supervisor
+func (s *SupervisorService) CreateOrUpdate(spec interface{}) (string, error) {
+	r, err := s.client.NewRequest("POST", supervisorPathPrefix, spec)
+	if err != nil {
+		return "", err
+	}
+	var result CreateSupervisorResponse
+	_, err = s.client.Do(r, &result)
+	if err != nil {
+		return "", err
+	}
+	return result.SupervisorId, nil
+}
+
+func applySupervisorId(input string, supervisorId string) string {
+	return strings.Replace(input, ":supervisorId", supervisorId, 1)
+}
+
+// GetActiveIDs returns array of active supervisor IDs
+// https://druid.apache.org/docs/latest/api-reference/supervisor-api/#get-an-array-of-active-supervisor-ids
+func (s *SupervisorService) GetActiveIDs() ([]string, error) {
+	return nil, status.Errorf(codes.Unimplemented, "method GetActiveIDs not implemented")
```

Comment:
Why to return status? As far as I can reckon this is the library, so any extras of status seem to be overkill and have little significance outside the scope of handling gRPC calls on the server or client side. 

In my eyes it only pulls in unnecessary dependency.

If the reason is only marking methods as then simple `errors.New("not implemented")` is as good, IMHO.

Reply:
I will fix that.